samples/ios/app/glean-sample-appUITests/DeletionRequestPingTest.swift

```diff
@@ -43,6 +43,11 @@ class DeletionRequestPingTest: XCTestCase {
         app.launchArguments = ["USE_MOCK_SERVER", "\(port)"]
         app.launch()
 
+        // Sleep for a 100ms to prevent an unlikely race condition that, while
+        // not practically acheiveable in real-world situations, does appear to
```

Comment:
What is this race condition? Can you describe what's racing against what and why that's unlikely to happen?

Reply:
Yeah, I was about to ask the same.
If we _are_ running into a race condition here we should try to plug that as well (even if that happens in a followup and we do this hack for now)

Reply:
The 'race' was more hypothesis than anything concrete.  I could never get this to reproduce locally so the only thing I could think of that might affect this is that the test toggles the telemetry enabled switch as soon as the app is launched which could potentially still be during Glean's initialization.  The race condition is I believe [here](https://searchfox.org/glean/source/glean-core/ios/Glean/Dispatchers.swift#205) where we are switching between queuing and not queuing when flushing tasks.  That's a pretty narrow windows, I admit, but the only other thing I could think of was that the `setUploadEnabled` was getting called _before_ init was finished and thus ran into [this](https://searchfox.org/glean/source/glean-core/ios/Glean/Glean.swift#231).

Reply:
> 
> 
> The 'race' was more hypothesis than anything concrete. I could never get this to reproduce locally so the only thing I could think of that might affect this is that the test toggles the telemetry enabled switch as soon as the app is launched which could potentially still be during Glean's initialization. 

Shouldn't this be covered by #1046?

>The race condition is I believe [here](https://searchfox.org/glean/source/glean-core/ios/Glean/Glean.swift#231), if the app beats Glean to here with toggling the switch before init is complete, then no deletion-request ping.

I believe we should probably let this race here then, and add more instrumentation (= log) to test the theories instead of suppressing it

Reply:
As said in a private chat looking at this again (and oh, I wrote that code), that should probably be an atomic to prevent a possible race I can think of.

Otherwise I agree: Can we add logging and see if we trigger it again on CI?

Reply:
I have no issues with that, it looks like we already log this as an error inside of glean but I don't see any indication of this in the test logs. The test logs don't appear to capture device logging.